tests/js/web/TableDetailComponentTest.js

```diff
@@ -115,4 +115,24 @@ describe('TableDetailComponent', () => {
 			expect($body.find('[data-table-detail-panel]').hasClass('table-detail--open')).to.be.false;
 		});
 	})
+
+	describe('When the backdrop overlay is clicked', () => {
+		it('should remove the "in" class from the backdrop to hide it', () => {
+			tableDetailComponent.init($body);
+
+			$body.find('[data-table-detail-view-detail]').trigger(clickEvent);
+			$body.find('.table-detail-backdrop').trigger(clickEvent);
+
+			expect($body.find('.table-detail-backdrop').hasClass('in')).to.be.false;
```

Comment:
Wouldn't this assertion be true if the `in` class was never added as part of the first triggered click?

could we perhaps make this more of an integration test with:

```
$body.find('[data-table-detail-view-detail]').trigger(clickEvent);

expect($body.find('.table-detail-backdrop').hasClass('in')).to.be.true;

$body.find('.table-detail-backdrop').trigger(clickEvent);

expect($body.find('.table-detail-backdrop').hasClass('in')).to.be.false;
```